apps/omg/lib/omg/state/transaction/payment.ex

```diff
@@ -80,11 +80,11 @@ defmodule OMG.State.Transaction.Payment do
   @doc """
   Transaform the structure of RLP items after a successful RLP decode of a raw transaction, into a structure instance
   """
-  def reconstruct([inputs_rlp, outputs_rlp | rest_rlp])
-      when rest_rlp == [] or length(rest_rlp) == 1 do
+  def reconstruct([inputs_rlp, outputs_rlp, tx_data_rlp, metadata_rlp]) do
     with {:ok, inputs} <- reconstruct_inputs(inputs_rlp),
          {:ok, outputs} <- reconstruct_outputs(outputs_rlp),
-         {:ok, metadata} <- reconstruct_metadata(rest_rlp),
+         :ok <- reconstruct_tx_data(tx_data_rlp),
```

Comment:
nitpick: since it's written this way, would it be better to just have this follow the same call style?

```elixir
{:ok, _tx_data} = reconstruct_tx_data(metadata_rlp)
```

Reply:
I was wondering what was less confusing, since we're disregarding the value as long as it's `0`. Maybe I'll just rename the `reconstruct` to `verify` in this case?

Reply:
@pdobacz comment was mostly for consistency with the call stack here. The outcome of what the code is clear, but in terms of aesthetics of the code, it would conform nicely if it followed the rest of the stack which is also doing:

```elixir
    with {:ok, inputs} <- reconstruct_inputs(inputs_rlp),
         {:ok, outputs} <- reconstruct_outputs(outputs_rlp),
         {:ok, metadata} <- reconstruct_metadata(rest_rlp),
         {:ok, _tx_data} <- reconstruct_tx_data(tx_data_rlp),
```

Reply:
if we're never caring about the output, I think it's better to not return it and have a representative function name.